apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -208,30 +208,38 @@ defmodule Aecore.Peers.Sync do
   end
 
   # Gets all unknown blocks, starting from the given one
-  defp check_peer_block(peer_uri, block_hash, state) do
-    case Chain.has_block?(block_hash) do
-      false ->
-        case(HttpClient.get_block({peer_uri, block_hash})) do
-          {:ok, deserialized_block} ->
+  defp check_peer_blocks(peer_uri, from_block_hash, state) do
```

Comment:
We should preserve the `Chain.has_block?(block_hash)` to be sure we don't download any block twice.